catalog/src/test/java/it/eng/catalog/rest/api/DatasetAPIControllerTest.java

```diff
@@ -0,0 +1,80 @@
+package it.eng.catalog.rest.api;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import it.eng.catalog.rest.api.DatasetAPIController;
+import it.eng.catalog.service.DataSetService;
+import it.eng.catalog.util.MockObjectUtil;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.ResponseEntity;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class DatasetAPIControllerTest {
+
+    DatasetAPIController datasetAPIController;
+
+    @Mock
+    private DataSetService dataSetService;
+
+    @BeforeEach
+    public void init() {
+        datasetAPIController = new DatasetAPIController(dataSetService);
+    }
+
+
+    @Test
+    public void getDatasetByIdSuccessfulTest() {
+        when(dataSetService.getDataSetById(MockObjectUtil.DATASET.getId())).thenReturn(MockObjectUtil.DATASET);
+        ResponseEntity<JsonNode> response = datasetAPIController.getDatasetById(MockObjectUtil.DATASET.getId());
+
+        assertNotNull(response);
+        assertTrue(response.getStatusCode().is2xxSuccessful());
+        assertTrue(StringUtils.contains(response.getBody().toString(), MockObjectUtil.DATASET.getType()));
+    }
+
+    @Test
+    public void getAllDatasetsSuccessfulTest() {
+        when(dataSetService.getAllDataSets()).thenReturn(MockObjectUtil.DATASETS);
+        ResponseEntity<JsonNode> response = datasetAPIController.getAllDatasets();
+
+        assertNotNull(response);
+        assertTrue(response.getStatusCode().is2xxSuccessful());
+        assertTrue(StringUtils.contains(response.getBody().toString(), MockObjectUtil.DATASET.getType()));
+    }
+
+//    @Test
```

Comment:
Why is this and update test commented out?

Reply:
It needed some changes, which honestly I forgot, not it is uncommented and updated.